core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1383,6 +1383,8 @@ private[spark] class DAGScheduler(
                 if (!job.finished(rt.outputId)) {
                   job.finished(rt.outputId) = true
                   job.numFinished += 1
+                  taskScheduler.markPartitionIdAsCompletedAndKillCorrespondingTaskAttempts(
```

Comment:
seems it's no harm to do it for shuffle task as well?

Reply:
Makes sense, have updated the code.